x-pack/plugins/security_solution/public/common/components/filter_bar/index.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo, useCallback } from 'react';
+
+import type { DataViewBase, Filter, Query } from '@kbn/es-query';
+import { FilterManager, TimeHistory } from '@kbn/data-plugin/public';
+import { DataView } from '@kbn/data-views-plugin/public';
+import { SearchBar, SearchBarProps } from '@kbn/unified-search-plugin/public';
+import { Storage } from '@kbn/kibana-utils-plugin/public';
+
+export interface FilterBarComponentProps {
+  dataTestSubj?: string;
+  indexPattern: DataViewBase;
+  isLoading?: boolean;
+  isRefreshPaused?: boolean;
+  filterQuery: Query;
+  filterManager: FilterManager;
+  filters: Filter[];
+  refreshInterval?: number;
+  displayStyle?: SearchBarProps['displayStyle'];
+}
+
+export const FilterBar = memo<FilterBarComponentProps>(
+  ({
+    indexPattern,
+    isLoading = false,
+    isRefreshPaused,
+    filterQuery,
+    filterManager,
+    filters,
+    refreshInterval,
+    dataTestSubj,
+    displayStyle,
+  }) => {
+    const onFiltersUpdated = useCallback(
+      (newFilters: Filter[]) => {
+        filterManager.setFilters(newFilters);
+      },
+      [filterManager]
+    );
+
+    const indexPatterns = useMemo(() => [indexPattern], [indexPattern]);
```

Comment:
It looks like this can be redundant, because of

https://github.com/elastic/kibana/pull/132507/files#diff-a925160d1e210da2bf3fce96e042ced447e87f9bc1c1a9b54414515817c32f43R65

```typescript

        timeHistory={new TimeHistory(new Storage(localStorage))}

```

Probably worth memoize timeHistory as well

Reply:
True! Will also fix in https://github.com/elastic/kibana/blob/47de4e77486040148cf2e5f98e3abd67e8401803/x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx#L132 where I took this from

Reply:
done